### PR TITLE
Cypress/E2E: Fix tests for cloud (part 4)

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Fix tests for cloud.

1. Test metadata
    - Add `@not_cloud` and `@te_only` labels to easily sort out spec files and run against corresponding test server
2. Flags to fail fast
      - Add `cy.shouldNotRunOnCloudEdition`, `cy.shouldRunOnTeamEdition`, `cy.shouldHaveElasticsearchDisabled` and `cy.shouldHavePluginUploadEnabled` to fail fast whenever certain requirement is not met
3. Default config for on prem and cloud edition
      - Rename `partial_default_config.json` to `on_prem_default_config.json`, and move from `/fixtures` to `/support/api` folder 
      - Add `cloud_default_config.json` to easily identify what are the config that can be update by Cypress against Cloud server
      - Add `expectConfigToBeUpdatable` to fail whenever a setting can't be updated which normally happened with Cloud server where not all config can be updated by a sysadmin.
4. Move spec files
    - move tests into its own files if it has specific requirements. Most of those are tests that can't be ran on Cloud, or only for team edition
5. Remove `cy.dbGetUser` in the global `before` hook which tends to throw an error connection when connecting/disconnecting for each spec files
6. Remove unnecessary update to config when particular setting is already in the default config.
7. Remove `cy.apiDeleteLicense` whenever a test requires no license (or simply TE only). Instead, it's replaced with flag `cy.shouldRunOnTeamEdition` to check whether a server is on team edition. The idea is to not dynamically add or remove license when running Cypress test but make the test server license status to remain as is all through out the entire test. With this we'll have separate test run for TE, Enterprise (or E20) and Cloud. Also, note that in Cloud, we should not delete the license.
8. Fix some typo, ID and API return values.

To follow: Will update Cypress E2E docs

Submitted on top of https://github.com/mattermost/mattermost-webapp/pull/7143

### CODE REVIEW

**Review by josephbaylon — APPROVED** [review]
LGTM

File: e2e/cypress/integration/bot_accounts/helpers.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {getRandomId} from '../../utils';
+import * as TIMEOUTS from '../../fixtures/timeouts';
+
+export function createBotInteractive(team, username = `bot-${getRandomId()}`) {
+    // # Visit the integrations
```
Comment (prapti) [inline e2e/cypress/integration/bot_accounts/helpers.js:8]:
```suggestion
    // # Visit the Integrations > Bot Accounts page
```

File: e2e/cypress/integration/bot_accounts/helpers.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    cy.visit(`/${team.name}/integrations/bots`);
+
+    // # Click add bot
+    cy.get('#addBotAccount').click();
+
+    // # fill+submit form
```
Comment (prapti) [inline e2e/cypress/integration/bot_accounts/helpers.js:14]:
```suggestion
    // # Fill and submit form
```

File: e2e/cypress/integration/bot_accounts/helpers.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    cy.get('#username').type(username);
+    cy.get('#displayName').type('Test Bot');
+    cy.get('#saveBot').click();
+
+    // * verify confirmation page
```
Comment (prapti) [inline e2e/cypress/integration/bot_accounts/helpers.js:19]:
```suggestion
    // * Verify confirmation page
```

File: e2e/cypress/integration/bot_accounts/helpers.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    cy.url({timeout: TIMEOUTS.ONE_MIN}).
+        should('include', `/${team.name}/integrations/confirm`).
+        should('match', /token=[a-zA-Z0-9]{26}/);
+
+    // * verify confirmation form/token
```
Comment (prapti) [inline e2e/cypress/integration/bot_accounts/helpers.js:24]:
```suggestion
    // * Verify confirmation form/token
```

**Review by prapti — APPROVED** [review]
Just minor suggestions. Overall it looks great, just wondering if the spec files should be named differently than `<feature>_1_spec` and `<feature>_2_spec` to avoid confusion especially for devs and contributors who will add tests later, to something that's more meaningful, for e.g. `<feature>_nc_spec` (for no-cloud) and `<feature>_te_spec` (for te-only). But of course can be an improvement in the future.
Thank you for such a detailed work Saturn!

File: e2e/cypress/integration/enterprise/guest_accounts/helpers.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+            Enable: featureFlag,
+            RestrictCreationToDomains: whitelistedDomains,
+        },
+        ServiceSettings: {
+            EnableEmailInvitations: emailInvitation,
+        },
+    });
+}
+
+export export function invitePeople(typeText, resultsCount, verifyText, channelName = 'Town Square', clickInvite = true) {
```
[the remark this answers is not in the document]
Comment (saturninoabril) [reply e2e/cypress/integration/enterprise/guest_accounts/helpers.js:17]: Wow, nice catch Prapti!

File: e2e/cypress/integration/bot_accounts/helpers.js (hunk 2 above)
Comment (saturninoabril) [reply e2e/cypress/integration/bot_accounts/helpers.js:14]: Thanks! Not paying attention much during my copy paste :)

File: e2e/cypress/integration/enterprise/guest_accounts/helpers.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (saturninoabril) [reply e2e/cypress/integration/enterprise/guest_accounts/helpers.js]: Done, thanks.

File: e2e/cypress/integration/bot_accounts/helpers.js (hunk 1 above)
Comment (saturninoabril) [reply e2e/cypress/integration/bot_accounts/helpers.js]: Done, thanks.

File: e2e/cypress/integration/bot_accounts/helpers.js (hunk 3 above)
Comment (saturninoabril) [reply e2e/cypress/integration/bot_accounts/helpers.js]: Done, thanks.

File: e2e/cypress/integration/bot_accounts/helpers.js (hunk 4 above)
Comment (saturninoabril) [reply e2e/cypress/integration/bot_accounts/helpers.js]: Done, thanks.

File: e2e/cypress/integration/bot_accounts/helpers.js (hunk 2 above)
Comment (saturninoabril) [reply e2e/cypress/integration/bot_accounts/helpers.js]: Done, thanks.

**Comment by saturninoabril** [comment]
Thanks Prapti and good suggestion, especially on naming convention. Please take a look again.

**Comment by furqanmlk** [comment]
All changes makes sense to me.
One question though, have we finalized the methods/steps those are specific to cloud ?
Are we going to have naming convention with **cloud** name so new people like will easily find cloud specific function.

**Review by furqanmlk — APPROVED** [review]

**Comment by saturninoabril** [comment]
Maybe we can have it as is? I'd suggest to discuss more in details in channel/meeting, then agree and document on naming convention/folder restructure.

**Pull request merged**